### PR TITLE
fix for preview link

### PR DESCRIPTION
**Pull request by kauanzito**
link preview was not working as it should (show url info above message)
here I bring a small solution for that, and also a better formatting of the file.

### CODE REVIEW

**Comment by LorranX** [comment]
🤠

**Comment by undefined-60** [comment]
Pelo menos aqui nunca vi funcionar a função de link preview, pelo contrario, quando iria enviar uma mensagem mencionando alguns membros, a menção não funcionava corretamente.

**Comment by kauanzito** [comment]
> Pelo menos aqui nunca vi funcionar a função de link preview, pelo contrario, quando iria enviar uma mensagem mencionando alguns membros, a menção não funcionava corretamente.

**Comment by kauanzito** [comment]
> Pelo menos aqui nunca vi funcionar a função de link preview, pelo contrario, quando iria enviar uma mensagem mencionando alguns membros, a menção não funcionava corretamente.


give me an example of how you are doing the sending, you are using the example: sock.sendMessage(jid, { text: text, mentions: [jids] }) ?

**Comment by undefined-60** [comment]
> 

uso assim:
sock.sendMessage(id, { text: text, contextInfo: { mentionedJid: [mentions] } });

**Comment by kauanzito** [comment]
> > 
> 
> uso assim: sock.sendMessage(id, { text: text, contextInfo: { mentionedJid: [mentions] } });

there is no longer this "mentionedJid" in the submission, and it is not necessary to apply the "contextInfo" to use the mentions, use the way I gave you as an example...

**Comment by CaRLos1227** [comment]
Que legal!

**Comment by LorranX** [comment]
QUANTO VC COBRA PARa mim dar aulA?

**Comment by LorranX** [comment]
Que bacana, como não pensei nisto antes, parabéns pela ideia

**Comment by LorranX** [comment]
well done, thx

**Comment by itmor** [comment]
god help

**Comment by viniciusgdr** [comment]
Changing the file formatting does not "solve" the problem. I'm using it perfectly and it's not having any problems. So I don't think changes should be made.

**Comment by viniciusgdr** [comment]
Maybe you haven't installed link-preview-js. Because the function requires the same.

Try installing by: npm i link-preview-js or yarn add link-preview-js

**Comment by CoderThi** [comment]
.

**Comment by kauanzito** [comment]
> Changing the file formatting does not "solve" the problem. I'm using it perfectly and it's not having any problems. So I don't think changes should be made.



> Changing the file formatting does not "solve" the problem. I'm using it perfectly and it's not having any problems. So I don't think changes should be made.

if you look closer you will see that the formatting is just an append, the real change was in the "urlInfo"

**Comment by kauanzito** [comment]
> Maybe you haven't installed link-preview-js. Because the function requires the same.
> 
> Try installing by: npm i link-preview-js or yarn add link-preview-js

we tried this way and it continued without showing preview link.

**Comment by CoderThi** [comment]
Many users are reporting on this issue in groups and communities, for some it works normally but for many it has been causing stress, I know people who chose to use external modules to achieve good performance, I think it is important to fix this error for us not need to go after external modules and adaptations😔

**Comment by kauanzito** [comment]
> Many users are reporting on this issue in groups and communities, for some it works normally but for many it has been causing stress, I know people who chose to use external modules to achieve good performance, I think it is important to fix this error for us not need to go after external modules and adaptations😔

this is real, and i bring this brief solution.

**Comment by kauanzito** [comment]
> I know that for some this module is working perfectly, but man even downloading and reevaluating the modules and updating it doesn't work, and that's annoying I waste the time I would have to develop new commands in adding a new external module and adapting it, "nobody deserve"

it's really quite annoying even having to install external modules to "generate" a preview link, which should work perfectly for 100% of users.

**and that's why I bring the solution for everyone.**

**Comment by viniciusgdr** [comment]
no explicit workaround was done in the file. Only the formatting was witnessed in it. Be clear when submitting a PR.

Example:
await sock.sendMessage(id, { text: 'Hi, this was sent using https://github.com/adiwajshing/baileys' })

**Comment by SheIITear** [comment]
Link preview works fine for me as well, like it does for @viniciusgdr. Quickly checking open issues, there doesn't seem to be any related to link preview not working. 

Also when making prs; make them clear. Don't just smash bunch of unnecessary formatting (which I don't think is needed) along your edit. It's a pain in the ass to review or even try to find what you tried to edit with the 'hint' you gave,

**Comment by kauanzito** [comment]
> no explicit workaround was done in the file. Only the formatting was witnessed in it. Be clear when submitting a PR.
> 
> Example: await sock.sendMessage(id, { text: 'Hi, this was sent using https://github.com/adiwajshing/baileys' })

Ok, soon I will close this pull to open another one with only the change, without the formatting of the file so as not to confuse other people as it confused you.

**Comment by kauanzito** [comment]
> ink preview works fine for me as well, like it does for @viniciusgdr. Quickly checking open issues, there doesn't seem to be any related to link preview not working.
> 
> Also when making prs; make them clear. Don't just smash bunch of unnecessary formatting (which I don't think is needed) along your edit. It's a pain in the ass to review or even try to find what you tried to edit with the 'hint' you gave.
> 
> Also please use English as not everyone knows your language.

@SheIITear is it good like this? 

> > no explicit workaround was done in the file. Only the formatting was witnessed in it. Be clear when submitting a PR.
> > Example: await sock.sendMessage(id, { text: 'Hi, this was sent using https://github.com/adiwajshing/baileys' })
> 
> Ok, soon I will close this pull to open another one with only the change, without the formatting of the file so as not to confuse other people as it confused you.

@SheIITear it's better this way?

**Comment by viniciusgdr** [comment]
There is no reason to change what is already working. Be clear. Submit Evidence, Clear Arguments. 

I use it on many occasions for several important things, no problems were seen in my tests in the last hour.

**Comment by undefined-60** [comment]
> Talvez você não tenha instalado link-preview-js. Porque a função requer o mesmo.
> 
> Tente instalar por: npm i link-preview-js ou yarn add link-preview-js

Eu instalei o link-preview e ainda assim não foi aqui.

**Comment by Dakauann** [comment]
So... All the major changes that you did were on the code formatting? Link preview works fine for me, how it also works for @SheIITear and @viniciusgdr .

**Comment by undefined-60** [comment]
> nesse navegador aqui tem a opção de traduzir a pagina não acho que o fato de usar o português implique em algo.

**Comment by undefined-60** [comment]
> Então... Todas as principais mudanças que você fez foram na formatação do código? A visualização do link funciona bem para mim, como também funciona para@SheIITeare@viniciusgdr.

Eu não vi uma mudança drástica no codigo que foi apresentado, acho que o problema é outra coisa, aqui, mesmo tendo o link-preview instalado não gera as informações do url na mensagem.

**Comment by Dakauann** [comment]
> > Então... Todas as principais mudanças que você fez foram na formatação do código? A visualização do link funciona bem para mim, como também funciona para@SheIITeare@viniciusgdr.
> 
> Eu não vi uma mudança drástica no codigo que foi apresentado, acho que o problema é outra coisa, aqui, mesmo tendo o link-preview instalado não gera as informações do url na mensagem.

Then it may be a problem in your code...

**Comment by undefined-60** [comment]
> > > Então... Todas as principais mudanças que você fez na formatação do código? A visualização do link funciona bem para mim, como também funciona para@SheIITeare@viniciusgdr.
> > 
> > 
> > Eu não vi uma mudança drástica no código que foi apresentado, acho que o problema é outra coisa, aqui, mesmo tendo o link-preview instalado não gera as informações do url na mensagem.
> 
> Então pode ser um problema no seu código...

Talvez possa ser, mais no caso, isso parece acontecer com outras pessoas

**Comment by kauanzito** [comment]
**Okay, I updated the pull.**

**Comment by Dakauann** [comment]
> **Okay, I updated the pull.**

You didn't changed anything m8

**Comment by kauanzito** [comment]
> ou didn't changed anything m8

wait, it's on "pending"...

**Comment by kauanzito** [comment]
[https://github.com/adiwajshing/Baileys/pull/2069/files](url)

**Comment by kauanzito** [comment]
[Click here](https://github.com/adiwajshing/Baileys/pull/2069/files)

**Comment by SheIITear** [comment]
> > I know that for some this module is working perfectly, but man even downloading and reevaluating the modules and updating it doesn't work, and that's annoying I waste the time I would have to develop new commands in adding a new external module and adapting it, "nobody deserve"
> 
> 
> 
> it's really quite annoying even having to install external modules to "generate" a preview link, which should work perfectly for 100% of users.
> 
> 
> 
> **and that's why I bring the solution for everyone.**


So I finally was able to find what you had changed. You're still required to install a module to create the preview.

Also no. It shouldn't work 100% to **everyone**. Ever heard of bloat? If you want to use x you install x. But if you don't use y you shouldn't have to install y.

**Comment by kauanzito** [comment]
> > > I know that for some this module is working perfectly, but man even downloading and reevaluating the modules and updating it doesn't work, and that's annoying I waste the time I would have to develop new commands in adding a new external module and adapting it, "nobody deserve"
> > 
> > 
> > it's really quite annoying even having to install external modules to "generate" a preview link, which should work perfectly for 100% of users.
> > **and that's why I bring the solution for everyone.**
> 
> So I finally was able to find what you had changed. You're still required to install a module to create the preview.
> 
> Also no. It shouldn't work 100% to **everyone**. Ever heard of bloat? If you want to use x you install x. But if you don't use y you shouldn't have to install y.

what would you do if you were among those users that even with link-preview-js installed, it's not working?

**Comment by mmhospedagem** [comment]
A solution and changing the way of sending does not need to change the script


`await this.instance.socket?.sendMessage(jid, {

        forward: {
          key: {
            fromMe: true,
            remoteJid: String(this.instance.socket?.user!.id)
          },
          message: {

            extendedTextMessage: {
              text: "Venha conhecer a MMHospedagem",
              matchedText: "https://www.mmhospedagem.com.br",
              canonicalUrl: "https://www.mmhospedagem.com.br",
              description: "Promoção 50% OFF",
              title: "MMHospedagem",
              previewType: "image",
              jpegThumbnail: URL IN BASE64
            }

          }
        }

      }`
      
      I currently do it this way and it works!

**Comment by undefined-60** [comment]
> Uma solução e alterando a forma de envio não precisa alterar o script
> 
> ` wait this.instance.socket?.sendMessage(jid, {
> 
> ```
>     forward: {
>       key: {
>         fromMe: true,
>         remoteJid: String(this.instance.socket?.user!.id)
>       },
>       message: {
> 
>         extendedTextMessage: {
>           text: "Venha conhecer a MMHospedagem",
>           matchedText: "https://www.mmhospedagem.com.br",
>           canonicalUrl: "https://www.mmhospedagem.com.br",
>           description: "Promoção 50% OFF",
>           title: "MMHospedagem",
>           previewType: "image",
>           jpegThumbnail: URL IN BASE64
>         }
> 
>       }
>     }
> ```
> 
> } ´ Atualmente faço assim e funciona!

Parece bastante eficiente para min, qualquer coisa eu mesmo penso em alguma solução a parte por aqui

**Comment by undefined-60** [comment]
Eu optei por usar uma versão mais antiga (da parte onde gera a preview no messages.js (JavaScript))

    if ('text' in message) {
        const extContent = { ...message };
        if (!!options.getUrlInfo && message.text.match(Defaults_1.URL_REGEX)) {
            try {
                const data = await options.getUrlInfo(message.text);
                extContent.canonicalUrl = data['canonical-url'];
                extContent.matchedText = data['matched-text'];
                extContent.jpegThumbnail = data.jpegThumbnail;
                extContent.description = data.description;
                extContent.title = data.title;
                extContent.previewType = 0;
            }
            catch (error) { // ignore if fails
                (_a = options.logger) === null || _a === void 0 ? void 0 : _a.warn({ trace: error.stack }, 'url generation failed');
            }
        }
        m.extendedTextMessage = extContent;
    } (padrão)

Porém notei que, quando uma mensagem vem com algum tipo de texto a não ser o link a mensagem de resposta não recebe o preview. Para resolver tive apenas que ir em "link-preview.js" e mudar um pequeno detalhe: "'matched-text': info.url" por padrão virá text onde agora está o info.url.

**Comment by undefined-60** [comment]
> Eu optei por usar uma versão mais antiga (da parte onde gera a preview no messages.js (JavaScript))
> 
> ```
> if ('text' in message) {
>     const extContent = { ...message };
>     if (!!options.getUrlInfo && message.text.match(Defaults_1.URL_REGEX)) {
>         try {
>             const data = await options.getUrlInfo(message.text);
>             extContent.canonicalUrl = data['canonical-url'];
>             extContent.matchedText = data['matched-text'];
>             extContent.jpegThumbnail = data.jpegThumbnail;
>             extContent.description = data.description;
>             extContent.title = data.title;
>             extContent.previewType = 0;
>         }
>         catch (error) { // ignore if fails
>             (_a = options.logger) === null || _a === void 0 ? void 0 : _a.warn({ trace: error.stack }, 'url generation failed');
>         }
>     }
>     m.extendedTextMessage = extContent;
> } (padrão)
> ```
> 
> Porém notei que, quando uma mensagem vem com algum tipo de texto a não ser o link a mensagem de resposta não recebe o preview. Para resolver tive apenas que ir em "link-preview.js" e mudar um pequeno detalhe: "'matched-text': info.url" por padrão virá text onde agora está o info.url.

Funcionou perfeitamente por aqui, sem fazer grandes mudanças ou usar algo adicional.

**Comment by kauanzito** [comment]
> > Eu optei por usar uma versão mais antiga (da parte onde gera a preview no messages.js (JavaScript))
> > ```
> > if ('text' in message) {
> >     const extContent = { ...message };
> >     if (!!options.getUrlInfo && message.text.match(Defaults_1.URL_REGEX)) {
> >         try {
> >             const data = await options.getUrlInfo(message.text);
> >             extContent.canonicalUrl = data['canonical-url'];
> >             extContent.matchedText = data['matched-text'];
> >             extContent.jpegThumbnail = data.jpegThumbnail;
> >             extContent.description = data.description;
> >             extContent.title = data.title;
> >             extContent.previewType = 0;
> >         }
> >         catch (error) { // ignore if fails
> >             (_a = options.logger) === null || _a === void 0 ? void 0 : _a.warn({ trace: error.stack }, 'url generation failed');
> >         }
> >     }
> >     m.extendedTextMessage = extContent;
> > } (padrão)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Porém notei que, quando uma mensagem vem com algum tipo de texto a não ser o link a mensagem de resposta não recebe o preview. Para resolver tive apenas que ir em "link-preview.js" e mudar um pequeno detalhe: "'matched-text': info.url" por padrão virá text onde agora está o info.url.
> 
> Funcionou perfeitamente por aqui, sem fazer grandes mudanças ou usar algo adicional.

so... a user who had to use a piece of code from an old version to be able to generate preview link!

**Comment by undefined-60** [comment]
> > > Eu optei por usar uma versão mais antiga (da parte onde gera a preview no messages.js (JavaScript))
> > > ```
> > > if ('text' in message) {
> > >     const extContent = { ...message };
> > >     if (!!options.getUrlInfo && message.text.match(Defaults_1.URL_REGEX)) {
> > >         try {
> > >             const data = await options.getUrlInfo(message.text);
> > >             extContent.canonicalUrl = data['canonical-url'];
> > >             extContent.matchedText = data['matched-text'];
> > >             extContent.jpegThumbnail = data.jpegThumbnail;
> > >             extContent.description = data.description;
> > >             extContent.title = data.title;
> > >             extContent.previewType = 0;
> > >         }
> > >         catch (error) { // ignore if fails
> > >             (_a = options.logger) === null || _a === void 0 ? void 0 : _a.warn({ trace: error.stack }, 'url generation failed');
> > >         }
> > >     }
> > >     m.extendedTextMessage = extContent;
> > > } (padrão)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Porém note que, quando uma mensagem vem com algum tipo de texto a não ser o link a mensagem de resposta não recebe o preview. Para resolver apenas que ir em "link-preview.js" e alterar um pequeno detalhe: "'matched-text': info.url" por padrão virá texto onde agora está o info.url.
> > 
> > 
> > Funcionou por aqui, sem fazer grandes alterações ou usar algo.
> 
> então... um usuário que teve que usar um pedaço de código de uma versão antiga para poder gerar o link de visualização!

Bom, tem várias maneiras de gerar o link-preview essa é só uma delas. Podem haver soluções melhores, de qualquer modo é opcional do usuário usar ou não ou ele mesmo tentar uma solução a parte.

**Comment by kauanzito** [comment]
> > > > Eu optei por usar uma versão mais antiga (da parte onde gera a preview no messages.js (JavaScript))
> > > > ```
> > > > if ('text' in message) {
> > > >     const extContent = { ...message };
> > > >     if (!!options.getUrlInfo && message.text.match(Defaults_1.URL_REGEX)) {
> > > >         try {
> > > >             const data = await options.getUrlInfo(message.text);
> > > >             extContent.canonicalUrl = data['canonical-url'];
> > > >             extContent.matchedText = data['matched-text'];
> > > >             extContent.jpegThumbnail = data.jpegThumbnail;
> > > >             extContent.description = data.description;
> > > >             extContent.title = data.title;
> > > >             extContent.previewType = 0;
> > > >         }
> > > >         catch (error) { // ignore if fails
> > > >             (_a = options.logger) === null || _a === void 0 ? void 0 : _a.warn({ trace: error.stack }, 'url generation failed');
> > > >         }
> > > >     }
> > > >     m.extendedTextMessage = extContent;
> > > > } (padrão)
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Porém note que, quando uma mensagem vem com algum tipo de texto a não ser o link a mensagem de resposta não recebe o preview. Para resolver apenas que ir em "link-preview.js" e alterar um pequeno detalhe: "'matched-text': info.url" por padrão virá texto onde agora está o info.url.
> > > 
> > > 
> > > Funcionou por aqui, sem fazer grandes alterações ou usar algo.
> > 
> > 
> > então... um usuário que teve que usar um pedaço de código de uma versão antiga para poder gerar o link de visualização!
> 
> Bom, tem várias maneiras de gerar o link-preview essa é só uma delas. Podem haver soluções melhores, de qualquer modo é opcional do usuário usar ou não ou ele mesmo tentar uma solução a parte.

yes, i understand your side very well.
but wouldn't it be better with the lib itself (Baileys) providing this function working perfectly?

**Comment by kauanzito** [comment]
I remember that in version 3.5.3 I didn't have to worry about these functions, they worked perfectly for everyone!

**Comment by undefined-60** [comment]
> > > > > Eu optei por usar uma versão mais antiga (da parte onde gera a preview no messages.js (JavaScript))
> > > > > ```
> > > > > if ('text' in message) {
> > > > >     const extContent = { ...message };
> > > > >     if (!!options.getUrlInfo && message.text.match(Defaults_1.URL_REGEX)) {
> > > > >         try {
> > > > >             const data = await options.getUrlInfo(message.text);
> > > > >             extContent.canonicalUrl = data['canonical-url'];
> > > > >             extContent.matchedText = data['matched-text'];
> > > > >             extContent.jpegThumbnail = data.jpegThumbnail;
> > > > >             extContent.description = data.description;
> > > > >             extContent.title = data.title;
> > > > >             extContent.previewType = 0;
> > > > >         }
> > > > >         catch (error) { // ignore if fails
> > > > >             (_a = options.logger) === null || _a === void 0 ? void 0 : _a.warn({ trace: error.stack }, 'url generation failed');
> > > > >         }
> > > > >     }
> > > > >     m.extendedTextMessage = extContent;
> > > > > } (padrão)
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > Porém note que, quando uma mensagem vem com algum tipo de texto a não ser o link a mensagem de resposta não recebe o preview. Para resolver apenas que ir em "link-preview.js" e alterar um pequeno detalhe: "'matched-text': info.url" por padrão virá texto onde agora está o info.url.
> > > > 
> > > > 
> > > > Funcionou por aqui, sem fazer grandes mudanças ou usar algo.
> > > 
> > > 
> > > ... usuário que teve que usar antigamente um pedaço de código de uma para poder gerar ou então um link de visualização!
> > 
> > 
> > Bom, tem várias maneiras de gerar o link-preview essa é só uma delas. Podem haver soluções melhores, de qualquer modo é opcional do usuário ou não ele mesmo tentar uma solução a parte.
> 
> sim, eu entendo muito bem o seu lado. mas não seria melhor com a própria lib (Baileys) disponibilizando essa função funcionando perfeitamente?

Verdade, seria.

**Comment by viniciusgdr** [comment]
Before opening a PR, write in English. That way I, and the others, can understand more easily.

I'm using it normally and it's working. I see no reason to change it.
Make sure the site you want to preview actually has a preview image in it.

**Comment by undefined-60** [comment]
> Antes de abrir um PR, escreva em inglês. Assim eu, e os outros, podemos entender mais facilmente.
> 
> Estou usando normalmente e está funcionando. Não vejo razão para mudá-lo. Certifique-se de que o site que você deseja visualizar tenha realmente uma imagem de visualização.

Aqui tá funcionando mais não com o código padrão, tive que fazer umas alterações no meu caso

**Comment by kauanzito** [comment]
> Before opening a PR, write in English. That way I, and the others, can understand more easily.
> 
> I'm using it normally and it's working. I see no reason to change it. Make sure the site you want to preview actually has a preview image in it.

my tests were with the YouTube website (https://www.youtube.com)
which didn't generate any views before I made the modification that brought up this pull.

and as there are many reports that it is not working, I think it is super necessary that the change that makes the correction happen.

**Comment by undefined-60** [comment]
> > Antes de abrir um PR, escreva em inglês. Assim eu, e os outros, podemos entender mais facilmente.
> > Estou usando normalmente e está funcionando. Não vejo razão para mudá-lo. Certifique-se de que o site que você deseja visualizar tenha realmente uma imagem de visualização.
> 
> meus testes foram com o site do YouTube ( https://www.youtube.com ) que não gerou visualizações antes de fazer a modificação que trouxe esse pull.
> 
> e como há muitos relatos de que não está dando certo, acho super necessário que aconteça a mudança que faz a correção.

youtu.be também não gera visualização, mas youtube...watch sim.

**Comment by kauanzito** [comment]
> > > Antes de abrir um PR, escreva em inglês. Assim eu, e os outros, podemos entender mais facilmente.
> > > Estou usando normalmente e está funcionando. Não vejo razão para mudá-lo. Certifique-se de que o site que você deseja visualizar tenha realmente uma imagem de visualização.
> > 
> > 
> > meus testes foram com o site do YouTube ( https://www.youtube.com ) que não gerou visualizações antes de fazer a modificação que trouxe esse pull.
> > e como há muitos relatos de que não está dando certo, acho super necessário que aconteça a mudança que faz a correção.
> 
> youtu.be também não gera visualização, mas youtube...watch sim.

the tests were with preview-link-js, which did not generate any preview.
without any modification, which also did not generate a preview.
and with the modification you brought here, it generated perfectly.

**Comment by FatihArridho** [comment]
@adiwajshing

File: src/Utils/messages.ts [hunk 1]
```diff
@@ -281,11 +281,11 @@ export const generateWAMessageContent = async(
 
 		if(urlInfo) {
 			extContent.canonicalUrl = urlInfo['canonical-url']
-			extContent.matchedText = urlInfo['matched-text']
+			extContent.matchedText = urlInfo['canonical-url']
 			extContent.jpegThumbnail = urlInfo.jpegThumbnail
 			extContent.description = urlInfo.description
 			extContent.title = urlInfo.title
-			extContent.previewType = 0
+			extContent.previewType = 2
```
Comment (adiwajshing) [inline src/Utils/messages.ts:288]: what does this do exactly?

**Comment by itmor** [comment]
@kauanzito

**Comment by ewsbr** [comment]
Based on what @undefined-60, I figured out that the matchedText is being sent without https:
```json
{
    "message": {
        "extendedTextMessage": {
            "text": "https://url.com/params",
            "matchedText": "url.com/params",
            "canonicalUrl": "https://url.com/params",
            "description": "<desc>",
            "title": "<title>",
            "jpegThumbnail": "<thumbnail>"
        }
    }
}
```

By returning both matched-text and canonical-url as `info.url` in `getUrlInfo`, the link preview works properly and sends everything prefixed with http(s):
```js
return {
    'canonical-url': info.url,
    'matched-text': info.url,
    title: info.title,
    description: info.description,
    jpegThumbnail,
}
```

One thing to keep in mind, I'm not sure you're able to send link previews to someone who isn't in your contact list, as the URL simply becomes regular text. It seems to work fine in groups though (at least in Android 12).

Another problem is that websites like medium will be ignored in the link preview. The regex to exclude emails is simply `.*@.*`, so any URL with that character is ignored silently.
Sample URL: https://medium.com/@utkuu/error-handling-in-express-js-and-express-async-errors-package-639c91ba3aa2 

Why are emails ignored? Those seem to work fine as links and prompt the correct apps to open them with.

**Comment by itmor** [comment]
@kauanzito

**Comment by itmor** [comment]
@adiwajshing

**Comment by kauanzito** [comment]
> Based on what @undefined-60, I figured out that the matchedText is being sent without https:
> 
> ```json
> {
>     "message": {
>         "extendedTextMessage": {
>             "text": "https://url.com/params",
>             "matchedText": "url.com/params",
>             "canonicalUrl": "https://url.com/params",
>             "description": "<desc>",
>             "title": "<title>",
>             "jpegThumbnail": "<thumbnail>"
>         }
>     }
> }
> ```
> 
> By returning both matched-text and canonical-url as `info.url` in `getUrlInfo`, the link preview works properly and sends everything prefixed with http(s):
> 
> ```js
> return {
>     'canonical-url': info.url,
>     'matched-text': info.url,
>     title: info.title,
>     description: info.description,
>     jpegThumbnail,
> }
> ```
> 
> One thing to keep in mind, I'm not sure you're able to send link previews to someone who isn't in your contact list, as the URL simply becomes regular text. It seems to work fine in groups though (at least in Android 12).
> 
> Another problem is that websites like medium will be ignored in the link preview. The regex to exclude emails is simply `.*@.*`, so any URL with that character is ignored silently. Sample URL: https://medium.com/@utkuu/error-handling-in-express-js-and-express-async-errors-package-639c91ba3aa2
> 
> Why are emails ignored? Those seem to work fine as links and prompt the correct apps to open them with.

very well researched, can you test my form that I brought in this pull?

**Comment by ewsbr** [comment]
@kauanzito Your PR works fine for me. It does the same as returning url.info in `getUrlInfo`, but in the message generation function.

One problem I can't seem to solve is how to send "large" link previews. Setting previewType to 0, 1, 2 or 3 sends the same preview with the text in the same row. Changing the image size doesn't help either.

Here's an example of the preview differences: https://www.testingcatalog.com/content/images/size/w2000/2021/08/4eee605c-6846-4cb4-9552-b480dc43c311.jpeg

Does anyone know how to change the link preview size?

**Comment by kauanzito** [comment]
> @kauanzito Your PR works fine for me. It does the same as returning url.info in `getUrlInfo`, but in the message generation function.
> 
> One problem I can't seem to solve is how to send "large" link previews. Setting previewType to 0, 1, 2 or 3 sends the same preview with the text in the same row. Changing the image size doesn't help either.
> 
> Here's an example of the preview differences: https://www.testingcatalog.com/content/images/size/w2000/2021/08/4eee605c-6846-4cb4-9552-b480dc43c311.jpeg
> 
> Does anyone know how to change the link preview size?

externalAdReply

**Pull request closed without merging**